File: javascript/Search Insert Position.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var searchInsert = function(nums, target) {
    for( var i = 0; i <= nums.length - 1; i++ )
    {
        if( target <= nums[i] )
        {
          return i;
        }
    }
    return nums.length;
};

File: python3/Search Insert Position.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        startInd = 0
        endInd = len(nums)-1
        medianInd = 0
        while (startInd <= endInd):
            medianInd = startInd + (endInd-startInd)//2
            
            if nums[medianInd] == target:
                return medianInd
            elif nums[medianInd] > target:
                endInd = medianInd - 1
            else:
                startInd = medianInd + 1
        
        return startInd
        